lib/tbot/config/template_kubernetes.go

```diff
@@ -226,3 +183,51 @@ func (t *templateKubernetes) render(
 
 	return trace.Wrap(destination.Write(ctx, defaultKubeconfigPath, yamlCfg))
 }
+
+// selectKubeConnectionMethod determines the address and SNI that should be
+// put into the kubeconfig file.
+func selectKubeConnectionMethod(proxyPong *webclient.PingResponse) (clusterAddr string, sni string, err error) {
+	// First we check for TLS routing. If this is enabled, we use the Proxy's
+	// PublicAddr, and we must also specify a special SNI.
+	//
+	// Even if KubePublicAddr is specified, we still use the general
+	// PublicAddr when using TLS routing.
+	if proxyPong.Proxy.TLSRoutingEnabled {
+		addr := proxyPong.Proxy.SSH.PublicAddr
+		host, _, err := net.SplitHostPort(proxyPong.Proxy.SSH.PublicAddr)
+		if err != nil {
+			return "", "", trace.Wrap(err, "parsing proxy public_addr")
+		}
+
+		sni = fmt.Sprintf("%s%s", constants.KubeTeleportProxyALPNPrefix, host)
+		hostIsIP := net.ParseIP(host) != nil
+		if host == "" || hostIsIP {
+			sni = fmt.Sprintf("%s%s", constants.KubeTeleportProxyALPNPrefix, constants.APIDomain)
```

Comment:
```suggestion
			sni = constants.KubeTeleportProxyALPNPrefix+ constants.APIDomain
```

Using `Sprintf` to concatenate 2 strings is kind of overkill, no?

Reply:
Just use `client.GetKubeTLSServerName`

Reply:
Good shout @tigrato - 8bbff66eb2